torch/_subclasses/fake_tensor.py

```diff
@@ -137,6 +137,8 @@ def from_real_tensor(self, fake_mode, t):
             out = FakeTensor(fake_mode, self.meta_converter(t), existing_device)
         if type(t) is torch.nn.Parameter:
             out = torch.nn.Parameter(out, requires_grad=out.requires_grad)  # type: ignore[assignment]
+        if safe_is_leaf(t) and t.grad is not None:
+            out.grad = self.from_real_tensor(fake_mode, t.grad)
```

Comment:
what if we have grad on a non-leaf tensor 🤔

Reply:
TBH, i'm not really sure...

Reply:
I think I'd just drop the leaf check

```
>>> torch.zeros(2, requires_grad=True)
tensor([0., 0.], requires_grad=True)
>>> x = torch.zeros(2, requires_grad=True)
>>> y = x * 2
>>> y.grad = torch.zeros(2)
>>> y
tensor([0., 0.], grad_fn=<MulBackward0>)
>>> y.grad
tensor([0., 0.])
```